sql/core/src/test/scala/org/apache/spark/sql/StreamTest.scala

```diff
@@ -179,16 +206,38 @@ trait StreamTest extends QueryTest with Timeouts {
          |${if (streamDeathCause != null) stackTraceToString(streamDeathCause) else ""}
          |
          |== Sink ==
-         |$sink
+         |${sink.toLongString}
          |
          |== Plan ==
          |${if (currentStream != null) currentStream.lastExecution else ""}
-         """
+         """.stripMargin
+
+    def assert(condition: => Boolean, message: String): Unit = {
```

Comment:
These are a bunch of changes to make sure that any kind of `assert` and `eventually` that is needed within `testStream` is wrapped in a try catch so that we can catch them and enrich the message with the `testState`.